libindi/drivers/auxiliary/lpm.cpp

```diff
@@ -0,0 +1,354 @@
+    IUFillText(&RecordFileT[1], "RECORD_FILE_NAME", "Name", "lpmlog.txt");
+    IUFillTextVector(&RecordFileTP, RecordFileT, NARRAY(RecordFileT), getDeviceName(), "RECORD_FILE", "Record File",
+                     MAIN_CONTROL_TAB, IP_RW, 0, IPS_IDLE);
+
+    // Unit Info
+    IUFillNumber(&UnitInfoN[0], "Calibdata", "", "%6.2f", -20, 30, 0, 0);
+    IUFillNumberVector(&UnitInfoNP, UnitInfoN, 1, getDeviceName(), "Unit Info", "",
+                       UNIT_TAB, IP_RO, 0, IPS_IDLE);
+
+    if (lpmConnection & CONNECTION_SERIAL)
+    {
+        serialConnection = new Connection::Serial(this);
+        serialConnection->registerHandshake([&]() { return getDeviceInfo(); });
+        serialConnection->setDefaultBaudRate(Connection::Serial::B_9600);
+        registerConnection(serialConnection);
+    }
+
+    addDebugControl();
+    addPollPeriodControl();
+
+    return true;
+}
+
+bool LPM::updateProperties()
+{
+    INDI::DefaultDevice::updateProperties();
+
+    if (isConnected())
+    {
+        defineNumber(&AverageReadingNP);
+        defineNumber(&UnitInfoNP);
+        defineSwitch(&ResetBP);
+        defineText(&RecordFileTP);
+        defineSwitch(&SaveBP);
+    }
+    else
+    {
+        deleteProperty(AverageReadingNP.name);
+        deleteProperty(UnitInfoNP.name);
+        deleteProperty(RecordFileTP.name);
+        deleteProperty(ResetBP.name);
+        deleteProperty(SaveBP.name);
+    }
+
+    return true;
+}
+
+bool LPM::ISNewText(const char *dev, const char *name, char *texts[], char *names[], int n)
+{
+    if (dev != nullptr && strcmp(dev, getDeviceName()) == 0)
+    {
+        if (strcmp(name, RecordFileTP.name) == 0)
+        {
+            IUUpdateText(&RecordFileTP, texts, names, n);
+            RecordFileTP.s = IPS_OK;
+            IDSetText(&RecordFileTP, nullptr);
+            return true;
+        }
+    }
+
+    return INDI::DefaultDevice::ISNewText(dev, name, texts, names, n);
+}
+bool LPM::ISNewSwitch(const char *dev, const char *name, ISState *states, char *names[], int n)
+{
+    char filename[2048];
+
+    if (dev != nullptr && strcmp(dev, getDeviceName()) == 0)
+    {
+        // Reset
+        if (!strcmp(name, ResetBP.name))
+        {
+            IUUpdateSwitch(&ResetBP, states, names, n);
+            ResetB[0].s = ISS_OFF;
+            ResetBP.s   = IPS_OK;
+            IDSetSwitch(&ResetBP, nullptr);
+
+            AverageReadingN[0].value = 0;
+            AverageReadingN[1].value = 0;
+            AverageReadingN[2].value = 0;
+            AverageReadingN[3].value = 0;
+
+            sumSQ = 0;
+            count = 0;
+            return true;
+        } else if (!strcmp(name, SaveBP.name))
+        {
+            IUUpdateSwitch(&SaveBP, states, names, n);
+            snprintf(filename, 2048, "%s/%s", RecordFileT[0].text, RecordFileT[1].text);
+
+            if (SaveB[SAVE_READINGS].s == ISS_ON)
+            {
+                LOGF_INFO("Save readings to %s", filename);
```

Comment:
what happens when the user changes the file name? shouldn't the previous file be closed a new one be opened?